### ISSUE TITLE
Optional in interface cause compilation error in generated Immutable.

### ISSUE THREAD

**Issue by danielesegato** [immutables/immutables#555]
I have an interface used by different Immutables and it cause compile time errors in the generated classes.

I semplified it to this:

```java
public interface ElementWithCaption {
    /**
     * @return element label / caption in plain text
     */
    // this is java 8 optional
    Optional<String> caption();
}
```

And the immutable:

```java
@Value.Immutable
public abstract class Test implements ElementWithCaption {
}
```

When I hit the compile button i get different errors  on `ImmutableTest`:

> Error:(36, 22) error: no suitable method found for equals(String,String)
> method Object.equals(Object) is not applicable
> (actual and formal argument lists differ in length)
> method ImmutableTest.equals(Object) is not applicable
> (actual and formal argument lists differ in length)

three times on this 3 generated methods:

```java
  public final ImmutableTest withCaption(String value) {
    String newValue = ImmutableTest.requireNonNull(value, "caption");
    if (ImmutableTest.equals(this.caption, newValue)) return this;
    return new ImmutableTest(newValue);
  }

    public final ImmutableTest withCaption(Optional<String> optional) {
    String value = optional.orElse(null);
    if (ImmutableTest.equals(this.caption, value)) return this;
    return new ImmutableTest(value);
  }

  private boolean equalTo(ImmutableTest another) {
    return ImmutableTest.equals(caption, another.caption);
  }
```

and this

> Error:(74, 34) error: no suitable method found for hashCode(String)
> method Object.hashCode() is not applicable
> (actual and formal argument lists differ in length)
> method ImmutableTest.hashCode() is not applicable
> (actual and formal argument lists differ in length)

on

```java
  @Override
  public int hashCode() {
    int h = 5381;
    h += (h << 5) + ImmutableTest.hashCode(caption);
    return h;
  }
```

am I missing something?

If I remove `Optional` the errors go away:

```java
public interface ElementWithCaption {
    /**
     * @return element label / caption in plain text
     */
    String caption();
}
```

is this a bug or am I doing something wrong / not supported? Couldn't find anything about immutable composition (through interfaces) in the documentation.

**Comment by elucash** [comment]
Thank you for reporting this. Apparently something strange is going on here, will take a look

**Comment by elucash** [comment]
released in 2.4.4. thank you!
